app/src/main/java/com/team/uptech/pomodoro/presentation/presenter/impl/MainPresenterImpl.kt

```diff
@@ -0,0 +1,117 @@
+package com.team.uptech.pomodoro.presentation.presenter.impl
+
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.team.uptech.pomodoro.dagger.scope.PerActivity
+import com.team.uptech.pomodoro.data.model.Pomodoro
+import com.team.uptech.pomodoro.domain.interactor.StartTimerUseCase
+import com.team.uptech.pomodoro.domain.interactor.TimerUseCase
+import com.team.uptech.pomodoro.presentation.presenter.MainPresenter
+import com.team.uptech.pomodoro.presentation.service.TimerService
+import com.team.uptech.pomodoro.presentation.ui.view.MainView
+import io.reactivex.android.schedulers.AndroidSchedulers
+import javax.inject.Inject
+
+/**
+ * Created on 27.04.17.
+ */
+@PerActivity
```

Comment:
why don't other dependencies have @PerActivity ? and your component doesn't have @PerActivity scope declared, so I'm not even sure how this will work. and yeah, you don't need an @Inject constructor if you have a declare injection in the module